server/src/main/java/org/elasticsearch/script/field/ToScriptField.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script.field;
+
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.elasticsearch.index.fielddata.ScriptDocValues.Dates;
+import org.elasticsearch.index.fielddata.ScriptDocValues.Doubles;
+import org.elasticsearch.index.fielddata.ScriptDocValues.Longs;
+import org.elasticsearch.index.fielddata.SortedNumericDoubleValues;
+
+public interface ToScriptField<T> {
+
+    DocValuesField<?> getScriptField(T docValues, String name);
+
+    ToScriptField<SortedNumericDocValues> _VERSION = (dv, n) -> new DelegateDocValuesField(new Longs(dv), n);
```

Comment:
I think I'd put VERSION and SEQNO directly on their field mappers

Reply:
I went ahead and moved all of these to their respective mappers as either method references or lambdas depending on whether or not a field had an appropriate constructor or is still relying on ScriptDocValues. I hadn't realized until this point that these are all singletons. (Duh!)